presto-jdbc/src/main/java/com/facebook/presto/jdbc/PrestoResultSet.java

```diff
@@ -1816,6 +1819,37 @@ private ResultsPageIterator(StatementClient client, Consumer<QueryStats> progres
         }
     }
 
+    private static class LengthLimitedIterator<T>
+            extends UnmodifiableIterator<T>
```

Comment:
This is not necessary in Java 8 now that `remove()` is a default method that throws